src/System.Runtime.Extensions/src/System/Net/WebUtility.cs

```diff
@@ -29,113 +31,152 @@ public static class WebUtility
 
         #region HtmlEncode / HtmlDecode methods
 
-        private static readonly char[] s_htmlEntityEndingChars = new char[] { ';', '&' };
-
         public static string HtmlEncode(string value)
         {
             if (string.IsNullOrEmpty(value))
             {
                 return value;
             }
 
-            // Don't create StringBuilder if we don't have anything to encode
-            int index = IndexOfHtmlEncodingChars(value, 0);
+            ReadOnlySpan<char> valueSpan = value.AsSpan();
+
+            // Don't create ValueStringBuilder if we don't have anything to encode
+            int index = IndexOfHtmlEncodingChars(valueSpan);
             if (index == -1)
             {
                 return value;
             }
 
-            StringBuilder sb = StringBuilderCache.Acquire(value.Length);
-            HtmlEncode(value, index, sb);
-            return StringBuilderCache.GetStringAndRelease(sb);
+            char[] rawBuffer = null;
+            Span<char> buffer = value.Length < 80 ?
+                stackalloc char[256] :
+                (rawBuffer = ArrayPool<char>.Shared.Rent(value.Length + 200));
```

Comment:
A comment on how these constants were chosen would be helpful.

Reply:
I added comments to explain why we chose these values. The fixed 200 additional characters could/should be transformed into a relative value but for that we first need more information on the distribution of input strings for HtmlEncode & HtmlDecode. See offline thread with Lee.

Reply:
I want to merge this in its current state as getting data might take some time. We can refine the buffer sizes afterwards.